src/order/category/Frame.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.category.Lattice
+import order.hom.complete_lattice
+
+/-!
+# The category of frames
+
+This file defines `Frame`, the category of frames.
+
+## References
+
+* [nLab, *Frm*](https://ncatlab.org/nlab/show/Frm)
+-/
+
+universes u
+
+open category_theory order
+
+/-- The category of frames. -/
+def Frame := bundled frame
+
+namespace Frame
+
+instance : has_coe_to_sort Frame Type* := bundled.has_coe_to_sort
+instance (X : Frame) : frame X := X.str
+
+/-- Construct a bundled `Frame` from a `frame`. -/
+def of (α : Type*) [frame α] : Frame := bundled.of α
+
+instance : inhabited Frame := ⟨of punit⟩
+
+instance : large_category.{u} Frame :=
+{ hom := λ X Y, frame_hom X Y,
+  id := λ X, frame_hom.id X,
+  comp := λ X Y Z f g, g.comp f,
+  id_comp' := λ X Y, frame_hom.comp_id,
+  comp_id' := λ X Y, frame_hom.id_comp,
+  assoc' := λ W X Y Z _ _ _, frame_hom.comp_assoc _ _ _ }
+
+instance : concrete_category Frame :=
+{ forget := ⟨coe_sort, λ X Y, coe_fn, λ X, rfl, λ X Y Z f g, rfl⟩,
+  forget_faithful := ⟨λ X Y, by convert fun_like.coe_injective⟩ }
```

Comment:
Can't you reuse more bundled machinery here?
```
instance : bundled_hom frame_hom := ...
```

Reply:
Unfortunately no, because `frame_hom` takes in `complete_lattice`, not `frame`. In general, I noticed that the `bundled` suite was pretty restrictive as it doesn't allow mixins and the typeclass for homs must match the typeclass for objects.

Reply:
Don't we already work around that type of problem when we bundle `monoid_hom`, by adding an abbreviation that takes the right arguments?

Reply:
Oh, I see that `Mon` uses this trick instead. But isn't a very bad idea to trade shortening one or two not-so-long instance declarations vs a permanent added layer to the already fragile hom hierarchy?

Reply:
It's not part of the heirarchy, it's an `abbreviation`:

https://github.com/leanprover-community/mathlib/blob/16ad25c20cf5b8be24357cf9e45bcb76ddbbcae0/src/algebra/category/Mon/basic.lean#L35-L46

Reply:
> Unfortunately no, because `frame_hom` takes in `complete_lattice`, not `frame`. In general, I noticed that the `bundled` suite was pretty restrictive as it doesn't allow mixins and the typeclass for homs must match the typeclass for objects.

And why is that? Shouldn't a `frame_hom` be between frames?

Reply:
Same answer as to "Shouldn't a `monoid_hom` be between monoids?" 😛

Reply:
Is this enough or do you want me to also use `bundled_hom.parent_projection` to deduce `has_forget₂ Frame Lattice`? It will be annoying because we don't have the instance `frame.to_lattice` (it's deduced transitively).